contracts/dezswap_pair/src/contract.rs

```diff
@@ -290,6 +273,34 @@ pub fn provide_liquidity(
         return Err(ContractError::InvalidZeroAmount {});
     }
 
+    // refund of remaining native token & desired of token
+    for (i, pool) in pools.iter().enumerate() {
+        let desired_amount = match total_share.is_zero() {
+            true => deposits[i],
+            false => pool.amount.multiply_ratio(share, total_share),
```

Comment:
decimal loss occurs here, eventually can decrease the share's value

Reply:
https://github.com/dezswap/dezswap-contracts/pull/4/commits/38b7400e024842aacd3d0c93452b51205e94ab02